src/v2/Components/FlashBanner/index.tsx

```diff
@@ -1,43 +1,65 @@
 import React from "react"
-import qs from "qs"
-import { Banner, Sans } from "@artsy/palette"
-
-/** Flash messages triggered by a url `flash_message` query param
- * A message can be a string or a React Component that takes no props
- * (that is, use relay or context if necessary)
- */
-const messageByQueryParam: { [k: string]: string | React.FC } = {
-  confirmed: "Your email has been confirmed.",
-  already_confirmed: "You have already confirmed your email.",
-  invalid_token: "An error has occurred. Please contact support@artsy.net.",
-  blank_token: "An error has occurred. Please contact support@artsy.net.",
-  expired_token: "Link expired. Resend verification email.",
-}
+import { Banner, Flex, Sans } from "@artsy/palette"
+import { QueryRenderer, graphql } from "react-relay"
+import { useSystemContext } from "v2/Artsy"
+import { messages, selectBannerMessage } from "./messages"
+import { FlashBannerProps } from "./types"
 
 /**
  * The component responsible for selecting a flash message key from the `flash_message` url query param
  */
-export const FlashBanner: React.FunctionComponent<{
-  messageCode?: string
-}> = props => {
-  const [messageCode, setMessageCode] = React.useState(props.messageCode)
+export const BaseFlashBanner: React.FunctionComponent<FlashBannerProps> = props => {
+  const [messageCode, setMessageCode] = React.useState<string | null>(
+    props.messageCode
+  )
 
   React.useEffect(() => {
+    // TODO: verify this won't cause a nasty bug where any incoming messageCode prop would break the messageCode logic
     if (!messageCode) {
-      const query = qs.parse(window.location.search.slice(1))
-      if (query["flash_message"]) setMessageCode(query["flash_message"])
+      const newCode = selectBannerMessage(props.me, window.location)
+
+      setMessageCode(newCode)
     }
-  }, [messageCode])
+  }, [props.me, messageCode])
 
   if (!messageCode) return null
-  const Message = messageByQueryParam[messageCode]
-  if (!Message) return null
+  const content = messages[messageCode]
 
-  return (
+  return content ? (
```

Comment:
Ternaries can be a bit obscure in terms of control flow. The pattern that's commonly seen is 

```tsx
if (!content) {
  return null
}

return (
  <Box>...</Box>
)
```